### PR TITLE
Plugins: Fix descendent frontend plugin signature validation

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:
Frontend plugins (or non-backend however you want to look at it), that are nested within a plugin, unless they have a MANIFEST.txt - will get marked as `unsigned` by default. Later in the validation process we check the current plugin's signature status, and since it's non backend we exit and don't update the status. Thanks to @aknuds1's [fix](https://github.com/grafana/grafana/issues/27364) for a related issue, we just need to move the check for plugin nesting sooner as this will recognize the plugin can inherit a signature, and should do so - preventing us from maintaining the `unsigned` status which shows up on the frontend

TLDR; It moves the logic for checking for root plugin signature earlier as currently it exits too soon while checking if the plugin is backend or not.

**Which issue(s) this PR fixes**:

Fixes #
https://github.com/grafana/grafana/issues/28629

### CODE REVIEW

File: pkg/plugins/plugins.go [hunk 1]
```diff
@@ -373,17 +373,7 @@ func (scanner *PluginScanner) IsBackendOnlyPlugin(pluginType string) bool {
 
 // validateSignature validates a plugin's signature.
 func (s *PluginScanner) validateSignature(plugin *PluginBase) *PluginError {
-	// For the time being, we choose to only require back-end plugins to be signed
-	// NOTE: the state is calculated again when setting metadata on the object
-	if !plugin.Backend || !s.requireSigned {
-		return nil
-	}
-
-	if plugin.Signature == PluginSignatureValid {
```
Comment (aknuds1) [inline pkg/plugins/plugins.go:382]: I think removing this is a bug, since the code line 392 will execute for plugins with valid signatures.

File: pkg/plugins/plugins.go [hunk 2]
```diff
@@ -373,17 +373,7 @@ func (scanner *PluginScanner) IsBackendOnlyPlugin(pluginType string) bool {
...
-		s.log.Debug("Plugin has valid signature", "id", plugin.Id)
-		return nil
-	}
-
+	// If the plugin has a parent plugin
```
Comment (aknuds1) [inline pkg/plugins/plugins.go:376]: I would remove this comment, since it's just overlapping with the one on 378.

**Review by aknuds1 — CHANGES_REQUESTED** [review]

**Review by dprokop — APPROVED** [review]
Tested and it works fine! Thanks @wbrowne

File: pkg/plugins/plugins.go (hunk 1 above)
Comment (wbrowne) [reply pkg/plugins/plugins.go:382]: It's not removed though, just moved. I essentially just moved the `if plugin.Root != nil {` block, but the diff comes out weird 😞

File: pkg/plugins/plugins.go [hunk 3]
```diff
@@ -403,6 +392,17 @@ func (s *PluginScanner) validateSignature(plugin *PluginBase) *PluginError {
...
+	// For the time being, we choose to only require back-end plugins to be signed
+	// NOTE: the state is calculated again when setting metadata on the object
+	if !plugin.Backend || !s.requireSigned {
+		return nil
+	}
+
+	if plugin.Signature == PluginSignatureValid {
+		s.log.Debug("Plugin has valid signature", "id", plugin.Id)
+		return nil
+	}
```
Comment (aknuds1) [inline pkg/plugins/plugins.go:404]: I think this one should be moved back, since the check for this case is [here](https://github.com/grafana/grafana/blob/master/pkg/plugins/plugins.go#L396).

**Review by aknuds1 — CHANGES_REQUESTED** [review]

**Review by aknuds1 — APPROVED** [review]
LGTM - tested and it works. Thanks!
